src/main/java/school/lemon/changerequest/java/generics/PairUtil.java

```diff
@@ -0,0 +1,53 @@
+package school.lemon.changerequest.java.generics;
+
+/**
+ * Created by lera on 20.12.16.
+ */
+public final class PairUtil {
+
+    public static <K, V> boolean equals(Pair<K, V> p1, Pair<K, V> p2) {
+        return p1.getKey().equals(p2.getKey()) && p1.getValue().equals(p2.getValue());
+    }
+
+    public static <K extends Comparable<K>, V> int compareTo(Pair<K, V> p1, Pair<K, V> p2) {
+        return p1.getKey().compareTo(p2.getKey());
+    }
+
+    public static <K, V> V[] getValues(Pair<K, V>[] pairs) {
+        V[] array = (V[]) new Object[pairs.length];
+        int i = 0;
+        for (Pair p : pairs) {
```

Comment:
It's better to use simple `for` loop instead of `for-each`. At least you should use `for-each` like `for (Pair<K, V> p : pairs)`.